contracts/messaging/MessageBusSender.sol

```diff
@@ -0,0 +1,126 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity 0.8.13;
+
+import "@openzeppelin/contracts-4.5.0/access/Ownable.sol";
+import "./interfaces/IGasFeePricing.sol";
+
+contract MessageBusSender is Ownable {
+    address public gasFeePricing;
+    uint64 public nonce;
+    uint256 internal fees;
+
+    constructor(address _gasFeePricing) {
+        gasFeePricing = _gasFeePricing;
+    }
+
+    event MessageSent(
+        address indexed sender,
+        uint256 srcChainID,
+        bytes32 receiver,
+        uint256 indexed dstChainId,
+        bytes message,
+        uint64 nonce,
+        bytes options,
+        uint256 fee,
+        bytes32 indexed messageId
+    );
+
+    function computeMessageId(
```

Comment:
A bit too picky, but you might want to enforce similar pattern in variable order both in events and functions.

Either of these, but not both at the same time:
- `srcChainId, srcAddress, dstChainId, dstAddress`
- `srcAddress, srcChainId, dstAddress, dstChainId`